services/provider/server/storageclaim.go

```diff
@@ -78,6 +70,15 @@ func (s *storageClassClaimManager) Create(ctx context.Context, consumer *ocsv1al
 			EncryptionMethod: encryptionMethod,
 		},
 	}
+	storageClassClaimObj.SetOwnerReferences([]metav1.OwnerReference{
```

Comment:
@nb-ohad am not sure it will work SetOwnerReferences is an inline function it just sets the ownerRef on object . this is same as before https://github.com/kubernetes/apimachinery/blob/v0.24.0/pkg/apis/meta/v1/meta.go#L172-L174

Reply:
I think it didn't work because of Kind and APIVersion issues, with the current change we are taking the information form the Schema and not the object (using GVK), this is the exact same logic that the controller utils function uses and should work